x-pack/plugins/security_solution/public/common/components/markdown_editor/plugins/index.ts

```diff
@@ -27,14 +26,30 @@ export const {
 
 export const platinumOnlyPluginTokens = [insightMarkdownPlugin.insightPrefix];
 
-export const uiPlugins = ({ licenseIsPlatinum }: { licenseIsPlatinum: boolean }) => {
+export const uiPlugins = ({
+  licenseIsPlatinum,
+  insightsUpsellingMessage,
```

Comment:
nit: I am wondering, if `uiPlugins` should get the `insightPluginWithLicense` or rather  all needed `pluginsWithLicense` as an array  so that `uiPlugins` do not have to care about `upsellingMessage` or `upsellingSections`.  Just thinking it out loud. It will just have the knowledge of `modified` plugins.

Reply:
I like the idea. The current design doesn't scale well if we need to add more license checks. but I don't think we should improve the `uiPlugins ` design on this PR. We are considering implementing ESS Upselling messages and components the same way we do for serverless. When it is implemented, this code will look simpler.

Reply:
@machadoum , makes sense. Thanks.